airflow/models/dagbag.py

```diff
@@ -398,6 +400,10 @@ def _process_modules(self, filepath, mods, file_last_changed_on_disk):
             dag.fileloc = mod.__file__
             try:
                 dag.timetable.validate()
+                # create a copy of params before validating
+                copied_params = copy.deepcopy(dag.params)
+                copied_params.update(conf or {})
```

Comment:
what is the value of `conf` here?

Reply:
Oops...didn't think much about that. Wrong here. I see that validating at create_dagrun is necessary too. I will add it back but then, since we validate during parsing, the check at create_dagrun would check for error when conf is added manually.

Reply:
@ephraimbuddy has this been fixed? From what I can tell `conf` is still not defined correctly, I may be missing something here.